iothub/service/tests/IotHubConnectionPropertiesTests.cs

```diff
@@ -18,6 +19,7 @@ public class IotHubConnectionPropertiesTests
         [DataRow("acme2.azure-devices.net", "acme2")]
         [DataRow("3acme.azure-devices.net", "3acme")]
         [DataRow("4-acme.azure-devices.net", "4-acme")]
+        [DataRow("5acme", "5acme")]
```

Comment:
this one is not really a IoT hub hostname?

Reply:
Do we also want to test if the (input) hostname is in valid hostname format (IOTHUB_NAME.azure-devices.net)?

Reply:
I think that could be a worthwhile thing to do, as the code currently implies that improper formats can be passed in as long as the user is sure that they are ok